### DISCUSSION TITLE
Dynamic configuration of `NEXTAUTH_URL` and `NEXTAUTH_SECRET`

### DISCUSSION THREAD

**Discussion by khteh** [nextauthjs/next-auth#7253] [Help]
### Question 💬

I have a React.JS/Next.JS application which can be accesed through 2 URLs and I configure it to use separate `clientId` and `clientSecret` based on `req.headers.host`.

I need to dynamically configure `NEXTAUTH_URL` based on the URL my application is accessed from. The DuendeIDS6Provider logs `"Invalid redirect_uri:` and 
```
export default async function auth (req: NextApiRequest, res: NextApiResponse) {
    await NextAuth(req, res, {
        callbacks: {
          async redirect ({ url, baseUrl }) {
            const origin = new URL(url).origin
            console.log(`redirect url: ${url}, baseUrl: ${baseUrl}, URL(url).origin: ${origin}`)
            return `${baseUrl}/management/overviews`
          }
        }
    });
});
```
shows redirecting to the base url configure in `NEXTAUTH_URL`.

How to configure `NEXTAUTH_URL` and `NEXTAUTH_SECRET` dynamically based on the same parameter at runtime?

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by cronin4392** [comment]
[You can set the secret as a parameter](https://next-auth.js.org/configuration/options#secret) to `NextAuth`. You can't however set `NEXTAUTH_URL`. I'm deploying on AWS Amplify and they do not expose preview deploy URLs and you have to build up yourself. It's super annoying to have to do all this logic and string concatenation in the `amplify.yml` file.

Please please please let us set NEXTAUTH_URL just like secret.

**Reply by andygirvan** [reply]
Did you manage to get this working even with the naming convention around preview environments? I had thought of getting the branch name but the preview environments are pr-23.etcetc.amplify.com which is not possible to get (I think??)

**Comment by martinmaillard** [comment]
Did anyone find a solution to this problem?

**Comment by mauops** [comment]
I'm facing the same issue

**Reply by rijk** [reply]
That’s what the ⬆️ upvote button is for.

**Comment by kiyosoft** [comment]
+1

**Comment by el-frontend** [comment]
Any update on this?

**Comment by nicks1m** [comment]
Why is this still not addressed?
